Makefile

```diff
@@ -18,26 +18,36 @@ VERSION = $(shell sed -n "s/<revision>\(.*\)<\/revision>/\1/p" $(PWD)/pom.xml |
 GIT_BRANCH := $(shell git rev-parse --abbrev-ref HEAD | sed s,^main$$,latest,g)
 GIT_HASH := $(shell git rev-parse HEAD)
 
+JAVA_VERSION_MAX_SUPPORTED=11
+JAVA_VERSION_MIN_SUPPORTED=11
 # Setting SHELL to bash allows bash commands to be executed by recipes.
 # This is a requirement for 'setup-envtest.sh' in the test target.
 # Options are set to exit when a recipe line exits non-zero or a piped command fails.
 SHELL = /usr/bin/env bash -o pipefail
 .SHELLFLAGS = -ec
-
-# check if maven is installed
-ifeq (,$(shell which $(MAVEN)))
-  $(error "No maven found in $(PATH). Please install maven")
-endif
-
-# check java version
-ifeq (,$(shell java --version))
-  $(error "No java found in $(PATH). Please install java >= 11")
-else
-  JAVA_VERSION=$(shell java --version | egrep -o "1[[:digit:]]\.[[:digit:]]+\.[[:digit:]]+" | head -n 1)
-    ifeq (,$(JAVA_VERSION))
-      $(error "Java version inferior to 11. Please install Java >= 11")
+mvn-checks:
+  # check if maven is installed
+  ifeq (,$(shell which $(MAVEN)))
+    $(error "No maven found in $(PATH). Please install maven")
+  else
+    MVN_JAVA_VERSION=$(shell mvn --version | sed -rn "s/.*Java version: ([[:digit:]]+)\.[[:digit:]]+\.[[:digit:]]+.*/\1/p")
+    MVN_VERSION_IS_SUPPORTED=$(shell [ $(MVN_JAVA_VERSION) -le $(JAVA_VERSION_MAX_SUPPORTED)  ] && [ $(MVN_JAVA_VERSION) -ge $(JAVA_VERSION_MIN_SUPPORTED)  ] && echo true)
+    ifeq ($(MVN_VERSION_IS_SUPPORTED), )
+      $(error "Maven Java $(MVN_JAVA_VERSION) version should be [$(JAVA_VERSION_MIN_SUPPORTED) ; $(JAVA_VERSION_MAX_SUPPORTED)]. Please use Java within these bounds with mvn")
+    endif
+  endif
+
+java-checks:
+  # check java version
+  ifeq (,$(shell java --version))
+    $(error "No java found in $(PATH). Please install java >= 11")
+  else
+    JAVA_VERSION=$(shell java --version | head -n 1 | sed -rn "s/.*\s([[:digit:]]+)\.[[:digit:]]+\.[[:digit:]]+\s.*/\1/p")
+    JAVA_VERSION_IS_SUPPORTED=$(shell [ $(JAVA_VERSION) -le $(JAVA_VERSION_MAX_SUPPORTED)  ] && [ $(JAVA_VERSION) -ge $(JAVA_VERSION_MIN_SUPPORTED)  ]&& echo true)
```

Comment:
it does not work with true because if ifeq. you need something to check agaist

Reply:
> bash is not yelling because of ]]&& echo true at the end? generally is very picky about this stuff
Nope not yelling, I guess it's because it's enclosed in `shell` but I'll add the space

> btw, why echo true and not just true
As @tupyy said, if I don't echo anything, nothing will be stored in the variable tested in `ifreq`, in plain bash I would test `$?`